### PR TITLE
feat: accept FlagsmithConfiguration in FlagsmithClient & small refactor

### PR DESCRIPTION
**Pull request by luk355**
feat: accept FlagsmithConfiguration in FlagsmithClient
fix: DI configuration of Example app to work in local evaluation mode
chore: tidy up namings of internal FlagsmithClient properties
chore: simplify code in HomeController

### CODE REVIEW

**Comment by matthewelwell** [comment]
Thanks @luk355! I will take a look at this next week!

File: Flagsmith.FlagsmithClient/Flagsmith.FlagsmithClient.csproj [hunk 1]
```diff
@@ -6,7 +6,7 @@
     <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
     <PackageId>Flagsmith</PackageId>
     <Title>Flagsmith</Title>
-    <Version>4.3.3</Version>
+    <Version>4.4.0</Version>
```
Comment (matthewelwell) [inline Flagsmith.FlagsmithClient/Flagsmith.FlagsmithClient.csproj:9]: Probably we can just mark this as 4.3.4 - I don't think we're adding extra functionality here that warrants a minor release?

**Review by matthewelwell — COMMENTED** [review]
Thanks for this @luk355, love all the refactoring! Could you look into why the actions are failing?

**Comment by luk355** [comment]
there were some whitespace characters left in the cs files. I've cleaned those up.

**Review by matthewelwell — APPROVED** [review]

**Comment by matthewelwell** [comment]
Thanks @luk355 - looks good to me. I'll merge once the actions pass and then get a release out.

**Comment by matthewelwell** [comment]
@luk355 sorry - looks like there are still some whitespace issues there, could you fix those up and I'll merge.

**Comment by luk355** [comment]
@matthewelwell, hopefully now  :)

**Pull request merged**